src/webview_provider/insightsPanel.ts

```diff
@@ -0,0 +1,465 @@
+        Uri.parse(params.projectRoot),
+      );
+      await workspace
+        .getConfiguration("dbt", workspaceFolder)
+        .update("deferConfigPerProject", newConfig, target);
+
+      if (syncRequestId) {
+        this._panel!.webview.postMessage({
+          command: "response",
+          args: {
+            syncRequestId,
+            body: {
+              updated: true,
+            },
+            status: true,
+          },
+        });
+      }
+    } catch (err) {
+      this.dbtTerminal.error(
+        "InsightsPanel",
+        "error while updating defer config",
+        err,
+      );
+      this._panel!.webview.postMessage({
+        command: "response",
+        args: {
+          syncRequestId,
+          body: {
+            updated: false,
+          },
+          status: false,
+        },
+      });
+    }
+  }
+
+  private async fetchProjectIntegrations(syncRequestId: string | undefined) {
+    try {
+      if (!this.altimateRequest.handlePreviewFeatures()) {
+        return;
+      }
+      this.dbtTerminal.debug("InsightsPanel", "Fetching project integrations");
+      const response = await this.altimateRequest.fetchProjectIntegrations();
+
+      if (!response?.length) {
+        this.dbtTerminal.debug("InsightsPanel", "Missing project integrations");
+        window
+          .showInformationMessage(
+            "You need to set up integration in SaaS. Please check the documentation",
+            ...["View", "Cancel"],
+          )
+          .then((selection) => {
+            if (selection === "View") {
+              env.openExternal(
+                Uri.parse("https://docs.myaltimate.com/test/defertoprod"),
+              );
+            }
+          });
+      }
+
+      if (syncRequestId) {
+        this._panel!.webview.postMessage({
+          command: "response",
+          args: {
+            syncRequestId,
+            body: response,
+            status: true,
+          },
+        });
+      }
+    } catch (err) {
+      this.dbtTerminal.error(
+        "InsightsPanel",
+        `could not fetch project integrations`,
+        err,
+      );
+      this._panel!.webview.postMessage({
+        command: "response",
+        args: {
+          syncRequestId,
+          body: {
+            response: [],
+          },
+          status: false,
+        },
+      });
+    }
+  }
+
+  private async testRemoteManifest(
+    syncRequestId: string | undefined,
+    dbtCoreIntegrationId: number,
+  ) {
+    try {
+      this.dbtTerminal.debug("InsightsPanel", "Fetching manifest signed url");
+      const response = await this.altimateRequest.fetchArtifactUrl(
+        "manifest",
+        dbtCoreIntegrationId,
+      );
+      // TODO: Can be removed once the backend sends a 404
+      if (response.url === "" && response.dbt_core_integration_file_id === -1) {
+        throw new Error(
+          "No remote manifest file present for selected dbt core integration",
+        );
+      }
```

Comment:
This should be handled through the exception thrown in altimate fetch method. Needs to be tested and better error message be displayed